ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml

```diff
@@ -0,0 +1,77 @@
+base_images:
+  ansible:
+    name: "4.10"
+    namespace: ocp
+    tag: ansible
+  cli:
+    name: "4.10"
+    namespace: ocp
+    tag: cli
+  dev-scripts:
+    name: test
+    namespace: ocp-kni
+    tag: dev-scripts
+  openstack-installer:
+    name: "4.10"
+    namespace: ocp
+    tag: openstack-installer
+  tools:
+    name: "4.10"
+    namespace: ocp
+    tag: tools
+  upi-installer:
+    name: "4.10"
+    namespace: ocp
+    tag: upi-installer
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.18
+releases:
+  initial:
```

Comment:
Hmm, seems like the initial release is not used.

Reply:
It's used.
```
time="2022-11-01T01:35:21Z" level=error error="failed to validate configuration /ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml: configuration has 3 errors:
  * tests[0].literal_steps.test[7].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured
  * tests[0].literal_steps.test[9].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured
  * tests[0].literal_steps.test[11].dependencies[0]: cannot determine source for dependency \"release:initial\" - this dependency requires a \"initial\" release, which is not configured"
```

Reply:
My bad. The latest release is not used should I say. Both initial and latest are the current version of the cluster. We can keep one.

```
yangyang:paused/ (master) $ grep -r RELEASE_IMAGE_LATEST                                                                      [9:50:27]
intermediate/cucushift-upgrade-paused-intermediate-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
latest/cucushift-upgrade-paused-latest-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
unpause/cucushift-upgrade-paused-unpause-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
```

Reply:
It's used, too.
```
time="2022-11-01T01:55:04Z" level=error error="failed to validate configuration /ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml: configuration has 6 errors:
  * tests[0].literal_steps.pre[7].dependencies[0]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.pre[7].dependencies[1]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[1].dependencies[0]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[7].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[9].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured
  * tests[0].literal_steps.test[11].dependencies[2]: cannot determine source for dependency \"release:latest\" - this dependency requires a \"latest\" release, which is not configured"
```

Reply:
Hmm, we need to remove the dependencies as well from the yaml files. They are not used in the shell scripts.
intermediate/cucushift-upgrade-paused-intermediate-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
latest/cucushift-upgrade-paused-latest-ref.yaml:    env: "RELEASE_IMAGE_LATEST"
unpause/cucushift-upgrade-paused-unpause-ref.yaml:    env: "RELEASE_IMAGE_LATEST"

Reply:
They are for the backward compatibility.  Some installation use RELEASE_IMAGE_LATEST, while some of the installation use RELEASE_IMAGE_INITIAL, to install the OCP cluster.

Reply:
Good point. I remember we discussed it in the unify CI sync meeting. But I can't remember the decision. Would it be using RELEASE_IMAGE_INITIAL first, if it's not present, using RELEASE_IMAGE_LATEST? @jianlinliu 
Most of our jobs don't specify `initial`. It's kind of weird to have both of them.

Reply:
I did some grep in the whole repo, only `ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh` is using `RELEASE_IMAGE_INITIAL`.
```
function getVersion() {
  local release_image=""
  if [ -n "${RELEASE_IMAGE_INITIAL-}" ]; then
    release_image=${RELEASE_IMAGE_INITIAL}
  elif [ -n "${RELEASE_IMAGE_LATEST-}" ]; then
    release_image=${RELEASE_IMAGE_LATEST}
  fi

  local version=""
  if [ ${release_image} != "" ]; then
    oc registry login
    version=$(oc adm release info ${release_image} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
  fi
  echo "${version}"
}
```
But it also support `RELEASE_IMAGE_LATEST`, it looks like it also works without `RELEASE_IMAGE_INITIAL`.  I am not sure if there would be some other backward compatibility issues.

Reply:
Anyway, it should not be a blocker for this PR. We can use a follow up PR to try to remove that (and related references in the steps). WDYT ?

Reply:
Sure.